packages/neos-ui-redux-store/src/CR/Nodes/index.ts

```diff
@@ -0,0 +1,474 @@
+    REMOVE = '@neos/neos-ui/CR/Nodes/REMOVE',
+    SET_STATE = '@neos/neos-ui/CR/Nodes/SET_STATE',
+    RELOAD_STATE = '@neos/neos-ui/CR/Nodes/RELOAD_STATE',
+    COPY = '@neos/neos-ui/CR/Nodes/COPY',
+    CUT = '@neos/neos-ui/CR/Nodes/CUT',
+    MOVE = '@neos/neos-ui/CR/Nodes/MOVE',
+    PASTE = '@neos/neos-ui/CR/Nodes/PASTE',
+    COMMIT_PASTE = '@neos/neos-ui/CR/Nodes/COMMIT_PASTE',
+    HIDE = '@neos/neos-ui/CR/Nodes/HIDE',
+    SHOW = '@neos/neos-ui/CR/Nodes/SHOW',
+    UPDATE_URI = '@neos/neos-ui/CR/Nodes/UPDATE_URI'
+}
+
+export type Action = ActionType<typeof actions>;
+
+/**
+ * Adds nodes to the application state. Completely *replaces*
+ * the nodes from the application state with the passed nodes.
+ *
+ * @param {Object} nodeMap A map of nodes, with contextPaths as key
+ */
+const add = (nodeMap: NodeMap) => createAction(actionTypes.ADD, {nodeMap});
+
+/**
+ * Adds/Merges nodes to the application state. *Merges*
+ * the nodes from the application state with the passed nodes.
+ *
+ * @param {Object} nodeMap A map of nodes, with contextPaths as key
+ */
+const merge = (nodeMap: NodeMap) => createAction(actionTypes.MERGE, {nodeMap});
+
+/**
+ * Marks a node as focused
+ *
+ * @param {String} contextPath The context path of the focused node
+ * @param {String} fusionPath The fusion path of the focused node, needed for out-of-band-rendering, e.g. when
+ *                            adding new nodes
+ */
+const focus = (contextPath: NodeContextPath, fusionPath: string) => createAction(actionTypes.FOCUS, {contextPath, fusionPath});
+
+/**
+ * Un-marks all nodes as not focused.
+ */
+const unFocus = () => createAction(actionTypes.UNFOCUS);
+
+/**
+ * Start node removal workflow
+ *
+ * @param {String} contextPath The context path of the node to be removed
+ */
+const commenceRemoval = (contextPath: NodeContextPath) => createAction(actionTypes.COMMENCE_REMOVAL, contextPath);
+
+/**
+ * Start node creation workflow
+ *
+ * @param {String} referenceNodeContextPath The context path of the referenceNode
+ * @param {String} referenceNodeFusionPath The fusion path of the referenceNode
+ */
+const commenceCreation = (referenceNodeContextPath: NodeContextPath, referenceNodeFusionPath: string) => createAction(actionTypes.COMMENCE_CREATION, {
+    referenceNodeContextPath,
+    referenceNodeFusionPath
+});
+
+/**
+ * Abort the ongoing node removal workflow
+ */
+const abortRemoval = () => createAction(actionTypes.REMOVAL_ABORTED);
+
+/**
+ * Confirm the ongoing removal
+ */
+const confirmRemoval = () => createAction(actionTypes.REMOVAL_CONFIRMED);
+
+/**
+ * Remove the node that was marked for removal
+ *
+ * @param {String} contextPath The context path of the node to be removed
+ */
+const remove = (contextPath: NodeContextPath) => createAction(actionTypes.REMOVE, contextPath);
+
+/**
+ * Set CR state on page load or after dimensions or workspaces switch
+ */
+const setState = (
+    {siteNodeContextPath, documentNodeContextPath, nodes, merge}: {
```

Comment:
Why not? We always do it everywhere. If feels like a named argument then, e.g. you don't have to memorize the arguments order etc. Anyways, since those action creators are semi-api, so we can't change them easily.

Reply:
No Problem, just asking for the intention 😉